xbmc/guilib/GUIFontCache.h

```diff
@@ -0,0 +1,296 @@
+/*!
+\file GUIFontCache.h
+\brief
+*/
+
+#ifndef CGUILIB_GUIFONTCACHE_H
+#define CGUILIB_GUIFONTCACHE_H
+#pragma once
+
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <cstddef>
+#include <cstring>
+#include <stdint.h>
+
+#include <algorithm>
+#include <vector>
+
+#include "boost/multi_index_container.hpp"
+#include "boost/multi_index/sequenced_index.hpp"
+#include "boost/multi_index/hashed_index.hpp"
+#include "boost/multi_index/member.hpp"
+#include "boost/shared_ptr.hpp"
```

Comment:
@bavison well i know i might come over annoying - but latest xcode which was released a couple of days ago chokes on those includes again. It seems the boost version we are using here on osx (1.44) doesn't really play well with c++11. Just fyi here is the printout ( you can't do anything about it - the problem is burried in one of the boost headers which gets implicitly included):

http://pastebin.com/ddQgCmdW

Can you think of any approach which could refactor this stuff without the usage of this boost stuff? ATM its not needed to support xcode 6.3 - but users will come and ask for support because it doesn't compile with latest osx build tools at some point ...